File: _08StringToInteger/Solution.go
package main

import (
	"fmt"
)

func myAtoi(str string) int {
	const max int = 0x7fffffff
	const min int = -0x80000000
	idx := 0
	for idx < len(str) && str[idx] == ' ' {
		idx ++
	}
	flag := true
	if idx < len(str) && str[idx] == '-' {
		flag = false
		idx ++
	} else if idx < len(str) && str[idx] == '+' {
		idx ++
	}
	ans := 0
	for idx < len(str) {
		if str[idx] > '9' || str[idx] < '0' {
			break
		}
		ans = ans * 10 + (int)(str[idx] - '0')
		idx ++
		if ans > max {
			if flag {
				return max
			} else {
				return min
			}
		}
	}
	if flag {
		return ans
	} else {
		return -ans
	}
}

func main() {
	a := myAtoi("-42")
	fmt.Println(a)
}

File: _08StringToInteger/Solution.js
/**
 * @param {string} str
 * @return {number}
 */
function myAtoi (str) {
    str = str.trim();
    let flag = true
    let idx = 0;
        max = 0x7fffffff;
        min = -0x80000000;
    if (str.charAt(idx) === '-') {
        flag = false;
        idx ++;
    } else if (str.charAt(idx) === '+') {
        idx ++;
    }

    let ans = 0
    while (idx < str.length) {
        if (str.charAt(idx) < '0' || str.charAt(idx) > '9') {
            break;
        }
        ans = ans*10 + parseInt(str.charAt(idx))
        idx ++;
        if (ans > max) {
            return flag ? max : min;
        }
    }
    if (flag) {
        return ans;
    } else {
        return -ans;
    }
}